apluslms_roman/observer.py

```diff
@@ -145,35 +147,77 @@ def result_msg(self, result):
         self._send_message(Message.RESULT_MSG, result.step, (result.code, result.error))
 
 
-ENTER_STATE_TEXTS = {
-    StepState.PREFLIGHT: "Pre-Flight tasks..",
-    StepState.PENDING: "Pending..",
-    StepState.RUNNING: "Running..",
-    StepState.POSTFLIGHT: "Post-Flight tasks..",
-    StepState.FAILED: "Failed!",
-    StepState.CANCELLED: "\rCancelled..",
-}
 class StreamObserver(BuildObserver):
-    def __init__(self, stream=None):
+    def __init__(self, stream=None, colors=True):
         super().__init__()
         self.stream = stream or sys.stdout
+        self.colors = colors
+        self.start_time = -1
+        init_color()
+
+    def write(self, to_write, colors=None):
+        if not isinstance(to_write, str):
+            to_write = str(to_write)
+        if self.colors:
+            if colors:
+                self.stream.write(colors)
+            self.stream.write(to_write + Style.RESET_ALL)
+        else:
+            self.stream.write(to_write)
 
     def _message(self, phase, type_, step=None, state=None, data=None):
-        fmt = "%s %s\n"
-        if type_ == Message.STATE_UPDATE:
-            data = ENTER_STATE_TEXTS.get(state)
-            if data is None:
+        def print_time(duration, succeeded=True):
+            duration = int(duration)
+            if succeeded:
+                self.write('ok', Fore.GREEN + Style.BRIGHT)
+            else:
+                self.write('failed', Fore.RED + Style.BRIGHT)
+            hours, seconds = divmod(duration, 3600)
```

Comment:
The time renderding could be a helper function, and then you can do stuff like `self.write("%s: %s" % (status_message, format_time(duration)), status_color)`

Reply:
Hmm.. Though the time is not colored... well any case.